pkg/agent/agent_readiness.go

```diff
@@ -81,7 +81,11 @@ func GetAllDesiredMembersAndArbitersPodState(namespacedName types.NamespacedName
 		p, err := podGetter.GetPod(podNamespacedName)
 		if err != nil {
 			if apiErrors.IsNotFound(err) {
+				// we can skip below iteration and check for our goal state since the pod is not available yet
 				podState.Found = false
+				podState.ReachedGoalState = false
+				podStates[i] = podState
```

Comment:
Alternatively (which reduces LoC), one can make a podName check here: https://github.com/mongodb/mongodb-kubernetes-operator/blob/2f089f9fb61c7c2e33dbba5465f294c6c78df031/pkg/agent/agent_readiness.go#L104-L108 but this only checks whether the podName is empty. Here we definitely know whether the error is not found.